lib/zspec/presenter.rb

```diff
@@ -134,6 +144,13 @@ def track_runtimes(results)
       }
     end
 
+    def track_runtimes_by_path(results)
+      key = (results.dig("summary", "file_path") || "").split("/", 4).take(3).join("/")
```

Comment:
```suggestion
      key = results.dig("summary", "file_path")&.split("/", 4)&.take(3)&.join("/") || ""
```
Would this be clearer?